publish.sh

```diff
@@ -1,10 +1,33 @@
 #!/usr/bin/env bash
 
+# This script will publish to rubygems and dockerhub, but only
+# if rubygems.org is not already up to date.
+
+. bash-lib/init
 set -eux
 
+published_gem_version="$(bl_gem_latest_version parse_a_changelog)"
+repo_gem_version="$(cat VERSION)"
+
+if [[ "${published_gem_version}" == "${repo_gem_version}" ]]; then
```

Comment:
usually we base it on whether the current commit matches the latest tag in the repo - not on whether the most recently published artifact doesn't match the VERSION file in repo (see https://github.com/cyberark/secretless-broker/blob/master/bin/publish#L36)

can we updated it here to use git tags as well? the ideal is that the code at the time of git tag was the exact code that was used to publish the artifact, so that someone looking on rubygems (or dockerhub, or wherever) can come back to the public repo and find the exact code that was used to build the public artifact

Reply:
If the pseudo code is:
```
if(master == latest_tag){
	publish()
}
```
Then publish will happen on every master build after a tag is created until another commit is pushed to master, which seems less than ideal. 

It's all hacks in my opinon because our Jenkins jobs are not triggering on tag creation, if we enabled that then we could just test if the trigger was a tag. Thats on my list but requires a refactor of how jobs are defined.

Reply:
You're not wrong, but I'm not sure I agree that in this one case we should employ a different hack than what we use everywhere else. It seems to me that logically the other hack makes more sense in terms of the desired end behavior, and if we continue to use the same tag-based hack everywhere, then the updates when we eventually improve how tags prompt pipelines to run will be more consistent

As you noted, it's effectively going to do the same thing here anyway, but I'd prefer to use the same type of logic everywhere and just fix that wherever it lives when we find a better way to do it. It doesn't make sense imo to have one project that operates differently.

Reply:
I disagree that this is effectively going to do the same thing. This method will prevent daily re-publishing to rubygems.org and dockerhub

My version of the desired end behaviour is that publication happens once for each release